src/main/scala/is/hail/check/Gen.scala

```diff
@@ -60,6 +65,32 @@ object Gen {
   def partition(rng: RandomDataGenerator, size: Int, parts: Int): Array[Int] =
     partition(rng, size, parts, (rng: RandomDataGenerator, avail: Int) => rng.nextInt(0, avail))
 
+  /**
+    * Takes {@code size} balls and places them into {@code parts} bins according
+    * to a dirichlet-multinomial distribution with alpha_n equal to {@code
+    * parts} for all n. The outputs of this function tend towards uniformly
+    * distributed balls, i.e. vectors close to the center of the simplex in
+    * {@code parts} dimensions.
+    *
+    **/
+  def partitionDirichlet(rng: RandomDataGenerator, size: Int, parts: Int): Array[Int] = {
+    val simplexVector = sampleDirichlet(rng, (0 until parts).map(x => parts.toDouble).toArray)
```

Comment:
Maybe `Array.fill[Double](parts)(parts)`?  Otherwise, use `_` instead of `x` if you don't need it.  This code definitely looks like a bug.